File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/Latest.java
package work.socialhub.kslack.entity;

import com.github.seratch.jslack.api.model.block.LayoutBlock;

import java.util.List;

public class Latest {

    private String clientMsgId;

    private String type;
    private String subtype;
    private String team;
    private String user;
    private String username;
    private String text;
    private String topic; // groups
    private List<Attachment> attachments;
    private List<LayoutBlock> blocks;
    private List<File> files;
    private List<Reaction> reactions;
    private Message.MessageRoot root;
    private boolean upload;
    private boolean displayAsBot;
    private String botId;
    private String botLink;
    private String threadTs;
    private String ts;
    private Message.Icons icons;
    private List<String> xFiles;

    public String getClientMsgId() {
        return this.clientMsgId;
    }

    public String getType() {
        return this.type;
    }

    public String getSubtype() {
        return this.subtype;
    }

    public String getTeam() {
        return this.team;
    }

    public String getUser() {
        return this.user;
    }

    public String getUsername() {
        return this.username;
    }

    public String getText() {
        return this.text;
    }

    public String getTopic() {
        return this.topic;
    }

    public List<Attachment> getAttachments() {
        return this.attachments;
    }

    public List<LayoutBlock> getBlocks() {
        return this.blocks;
    }

    public List<File> getFiles() {
        return this.files;
    }

    public List<Reaction> getReactions() {
        return this.reactions;
    }

    public Message.MessageRoot getRoot() {
        return this.root;
    }

    public boolean isUpload() {
        return this.upload;
    }

    public boolean isDisplayAsBot() {
        return this.displayAsBot;
    }

    public String getBotId() {
        return this.botId;
    }

    public String getBotLink() {
        return this.botLink;
    }

    public String getThreadTs() {
        return this.threadTs;
    }

    public String getTs() {
        return this.ts;
    }

    public Message.Icons getIcons() {
        return this.icons;
    }

    public List<String> getXFiles() {
        return this.xFiles;
    }

    public void setClientMsgId(String clientMsgId) {
        this.clientMsgId = clientMsgId;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setSubtype(String subtype) {
        this.subtype = subtype;
    }

    public void setTeam(String team) {
        this.team = team;
    }

    public void setUser(String user) {
        this.user = user;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public void setText(String text) {
        this.text = text;
    }

    public void setTopic(String topic) {
        this.topic = topic;
    }

    public void setAttachments(List<Attachment> attachments) {
        this.attachments = attachments;
    }

    public void setBlocks(List<LayoutBlock> blocks) {
        this.blocks = blocks;
    }

    public void setFiles(List<File> files) {
        this.files = files;
    }

    public void setReactions(List<Reaction> reactions) {
        this.reactions = reactions;
    }

    public void setRoot(Message.MessageRoot root) {
        this.root = root;
    }

    public void setUpload(boolean upload) {
        this.upload = upload;
    }

    public void setDisplayAsBot(boolean displayAsBot) {
        this.displayAsBot = displayAsBot;
    }

    public void setBotId(String botId) {
        this.botId = botId;
    }

    public void setBotLink(String botLink) {
        this.botLink = botLink;
    }

    public void setThreadTs(String threadTs) {
        this.threadTs = threadTs;
    }

    public void setTs(String ts) {
        this.ts = ts;
    }

    public void setIcons(Message.Icons icons) {
        this.icons = icons;
    }

    public void setXFiles(List<String> xFiles) {
        this.xFiles = xFiles;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/Latest.kt
package work.socialhub.kslack.entity

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import work.socialhub.kslack.entity.block.LayoutBlock
import work.socialhub.kslack.entity.bot.BotProfile
import work.socialhub.kslack.entity.file.File
import work.socialhub.kslack.entity.message.MessageIcons
import work.socialhub.kslack.entity.message.MessageRoot
import kotlin.js.JsExport

/**
 * Represents the latest message in a channel, including its metadata, content,
 * and associated objects like attachments, blocks, files, and reactions.
 *
 * Used in channel listing responses to provide a preview of the most recent
 * activity. Contains both the message content and flags indicating its type
 * (bot message, upload, thread, etc.).
 *
 * @see <a href="https://api.slack.com/types/message">Message type reference</a>
 */
@JsExport
@Serializable
class Latest {

    @SerialName("client_msg_id")
    var clientMsgId: String? = null

    var type: String? = null
    var subtype: String? = null
    var team: String? = null
    var user: String? = null
    var username: String? = null

    @SerialName("parent_user_id")
    var parentUserId: String? = null

    var text: String? = null
    var topic: String? = null // groups

    var attachments: Array<Attachment>? = null
    var blocks: Array<LayoutBlock>? = null
    var files: Array<File>? = null
    var reactions: Array<Reaction>? = null
    var root: MessageRoot? = null

    @SerialName("upload")
    var isUpload: Boolean = false

    @SerialName("display_as_bot")
    var isDisplayAsBot: Boolean = false

    @SerialName("app_id")
    var appId: String? = null

    @SerialName("bot_id")
    var botId: String? = null

    @SerialName("bot_link")
    var botLink: String? = null

    @SerialName("bot_profile")
    var botProfile: BotProfile? = null

    @SerialName("thread_ts")
    var threadTs: String? = null

    var ts: String? = null
    var icons: MessageIcons? = null

    @SerialName("x_files")
    var xFiles: Array<String>? = null

    var edited: Edited? = null
}
